### PR TITLE
distsql,store/tikv: SelectDAG function parameter refactor

### PR DESCRIPTION
**Pull request by tiancaiamao**
1. move some distsql.SelectDAG parameter to kv.Request struct
2. modify tikv.RPCContext struct, remove kvrpcpb.Context in it
3. let tikvrpc.Request struct share Context with its subcommand request

@coocood @disksing @hanfei1991

### CODE REVIEW

**Comment by coocood** [comment]
What's the benefit to share Context with its subcommand request?

**Comment by coocood** [comment]
LGTM

File: distsql/distsql.go [hunk 1]
```diff
@@ -244,10 +244,8 @@ func Select(client kv.Client, ctx goctx.Context, req *tipb.SelectRequest, keyRan
 }
 
 // SelectDAG sends a DAG request, returns SelectResult.
-// concurrency: The max concurrency for underlying coprocessor request.
-// keepOrder: If the result should returned in key order. For example if we need keep data in order by
-//            scan index, we should set keepOrder to true.
-func SelectDAG(client kv.Client, ctx goctx.Context, dag *tipb.DAGRequest, keyRanges []kv.KeyRange, concurrency int, keepOrder bool, desc bool, isolationLevel kv.IsoLevel, priority int) (SelectResult, error) {
+// In kvReq, KeyRanges is required, Concurrency/KeepOrder/Desc/IsolationLevel/Priority are optional.
+func SelectDAG(client kv.Client, ctx goctx.Context, kvReq *kv.Request) (SelectResult, error) {
```
Comment (zimulala) [inline distsql/distsql.go]: `SelectDAG(ctx goctx.Context, client kv.Client, kvReq *kv.Request)` is better.

File: executor/new_distsql.go [hunk 2]
```diff
@@ -133,9 +134,18 @@ func (e *TableReaderExecutor) Next() (Row, error) {
 
 // Open implements the Executor Open interface.
 func (e *TableReaderExecutor) Open() error {
-	kvRanges := tableRangesToKVRanges(e.tableID, e.ranges)
-	var err error
-	e.result, err = distsql.SelectDAG(e.ctx.GetClient(), goctx.Background(), e.dagPB, kvRanges, e.ctx.GetSessionVars().DistSQLScanConcurrency, e.keepOrder, e.desc, getIsolationLevel(e.ctx.GetSessionVars()), e.priority)
+	var builder requestBuilder
+	kvReq, err := builder.SetTableRanges(e.tableID, e.ranges).
+		SetDAGRequest(e.dagPB).
```
Comment (AndreMouche) [inline executor/new_distsql.go]: Could we init `kvReq` with `e` directly in one single function?

**Comment by zimulala** [comment]
LGTM

File: server/region_handler.go [hunk 3]
```diff
@@ -536,12 +536,12 @@ func (t *regionHandlerTool) getMvccByRecordID(tableID, recordID int64) (*kvrpcpb
...
 
 	tikvReq := &tikvrpc.Request{
-		Type:     tikvrpc.CmdMvccGetByKey,
-		Priority: kvrpcpb.CommandPri_Normal,
+		Type: tikvrpc.CmdMvccGetByKey,
 		MvccGetByKey: &kvrpcpb.MvccGetByKeyRequest{
 			Key: encodeKey,
 		},
 	}
+	tikvReq.Context.Priority = kvrpcpb.CommandPri_Normal
```
Comment (AndreMouche) [inline server/region_handler.go]: It seems  default priority  is `Normal`?

File: executor/new_distsql.go (hunk 2 above)
Comment (tiancaiamao) [reply executor/new_distsql.go]: They are nearly the same but not identical...such as `builder.SetTableRanges()` and `builder.SetTableHandles()`, so they can't be written into one single function.
Through it's not one single function, you can view it as a single line:

```
builder.SetXXX().SetYYY().SetZZZ().Build()
```

File: server/region_handler.go (hunk 3 above)
Comment (tiancaiamao) [reply server/region_handler.go]: Yes, this is a tiny refactor, the old code set it to `Normal` ...

**Pull request merged**